app/components/xkcd_mentions.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import annotations
+
+import asyncio
+import datetime as dt
+import re
+
+import discord
+import httpx
+from pydantic import BaseModel
+
+from app.utils import (
+    DeleteMessage,
+    MessageLinker,
+    TTRCache,
+    create_delete_hook,
+    create_edit_hook,
+    remove_view_after_timeout,
+)
+
+SECONDS_IN_HOUR = 3600
+XKCD_REGEX = re.compile(r"\bxkcd#(\d+)", re.IGNORECASE)
+XKCD_URL = "https://xkcd.com/{}"
+
+
+class XKCD(BaseModel):
+    day: int
+    month: int
+    year: int
+    img: str
+    title: str
+    alt: str
+
+
+class XKCDMentionCache(TTRCache[int, discord.Embed]):
+    async def fetch(self, key: int) -> None:
+        url = XKCD_URL.format(key)
+        async with httpx.AsyncClient() as client:
+            r = await client.get(f"{url}/info.0.json")
+        if not r.is_success:
+            error = (
+                f"XKCD #{key} does not exist."
+                if r.status_code == 404
+                else f"Unable to fetch XKCD #{key}."
+            )
+            self[key] = discord.Embed(color=discord.Color.red()).set_footer(text=error)
+            return
+
+        xkcd = XKCD(**r.json())
+        date = dt.datetime(
+            day=xkcd.day, month=xkcd.month, year=xkcd.year, tzinfo=dt.UTC
+        )
+        self[key] = (
+            discord.Embed(title=xkcd.title, url=url)
+            .set_image(url=xkcd.img)
+            .set_footer(text=f"{xkcd.alt} • {date:%B %-d, %Y}")
+        )
+
+
+xkcd_mention_cache = XKCDMentionCache(hours=12)
+xkcd_mention_linker = MessageLinker()
+
+
+class DeleteButton(DeleteMessage):
+    linker = xkcd_mention_linker
+    action_singular = "linked this XKCD comic"
+    action_plural = "linked these XKCD comics"
+
+
+async def xkcd_mention_message(
+    message: discord.Message,
+) -> tuple[list[discord.Embed], int]:
+    embeds = []
+    matches = list(dict.fromkeys(m[1] for m in XKCD_REGEX.finditer(message.content)))
+    omitted = None
+    if len(matches) > 10:
+        omitted = discord.Embed(color=discord.Color.orange()).set_footer(
+            text=f"{len(matches) - 9} XKCD comics were omitted."
+        )
+        # Nine instead of ten to account for the `omitted` embed.
+        matches = matches[:9]
+    tasks = (xkcd_mention_cache.get(int(m)) for m in matches)
+    embeds = await asyncio.gather(*tasks)
```

Comment:
Since this was simplified to be a gather with no outer list comp, we can drop the variable:
```suggestion
    embeds = await asyncio.gather(*(xkcd_mention_cache.get(int(m)) for m in matches))
```

Reply:
I was going to do that but I wasn't sure if I liked the `(*(`.